package/src/components/Channel/Channel.tsx

```diff
@@ -1207,6 +1207,7 @@ const ChannelWithContext = <
       await channel.query({}, 'latest');
     }
     await channel.state.loadMessageIntoState('latest');
+    setMessages([...channel.state.messages]);
```

Comment:
Whenever `latestMessages` was very short (for example if we have 7-8 unread messages), the list was cut off since the state was not being set properly.